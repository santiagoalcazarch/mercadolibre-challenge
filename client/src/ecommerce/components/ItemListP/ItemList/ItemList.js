
import React from 'react'
import PropTypes from 'prop-types'
import Item from '../Item/Item'
import { Link } from 'react-router-dom'
import { ITEM_FUNC } from '../../../../routes/routes'
import MessageBox from '../../../../shared/components/Message-box/Message_box'

/**
 * Permite mostrar la lista de items, cual se 
 * recibe como parametro
 * @param {array} itemList 
 * @returns 
 */
const ItemList = ({ itemList }) => {

  /**
   * Si la lista se encuentra vacía, despliega un mensaje
   */
  if ( itemList.length === 0 ) {
    return (
      <MessageBox
        title="¡Ups!"
        description="No hemos encontrado ningún articulo"
      />
    )
  }

  return (
    <div className="item-list-container">
      <div className="item-list-content">
        {
          itemList.map((item) =>
            <div className="item-list-item" key={item.id}>
              <Link to={ITEM_FUNC(item.id)}>
                <Item item={item} />
                <div className="item-divider"></div>
              </Link>
            </div>
          )
        }
      </div>
    </div>
  )
}

ItemList.propTypes = {
  itemList: PropTypes.array.isRequired
}

export default ItemList