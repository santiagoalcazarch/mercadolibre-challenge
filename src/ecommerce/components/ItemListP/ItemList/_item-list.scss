
.item-list-container{

  background-color: white;

  .item-list-content{
    
    .item-divider{
      height: 1px;
      background-color: $mainLightGray;
    }
  
    .item-divider:last-child{
      display: none;
    }

    .item-list-item{
      padding: 0 16px;
      transition: box-shadow 0.2s ease;
    }

    .item-list-item:hover{
      box-shadow: 0px 0px 4px rgba(0, 0, 0, 0.25);
    }
  }

}