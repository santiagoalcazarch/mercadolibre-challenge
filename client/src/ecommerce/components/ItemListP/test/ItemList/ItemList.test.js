
import { shallow } from "enzyme";
import ItemList from "../../ItemList/ItemList";
import { items, itemsEmpty } from "./itemList.mock";

describe('ItemList component test', () => { 

  const wrapper = shallow( <ItemList itemList={items} /> );

  test('Debería hacer match con el snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('Debería mostar todos los items', () => {
    // length validation
    const itemsMappged = wrapper.find(".item-list-item");
    expect(itemsMappged.length).toBe(items.length);

    // Keys validation
    const renderedItemsKeys = itemsMappged.map( item => item.key() );
    const baseItemsKeys = items.map( item => item.id );
    expect(renderedItemsKeys).toEqual(baseItemsKeys);    
  });

  test('Debería desplegar mensaje de error', () => {
    const wrapperListEmpty = shallow( <ItemList itemList={itemsEmpty} /> );
    const existMessage = wrapperListEmpty.find("MessageBox").exists();
    expect(existMessage).toBe(true);
  });

})