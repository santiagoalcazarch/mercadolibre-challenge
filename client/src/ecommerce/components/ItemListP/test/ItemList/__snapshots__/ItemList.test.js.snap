// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemList component test Debería hacer match con el snapshot 1`] = `
<div
  className="item-list-container animate__animated animate__fadeIn "
>
  <div
    className="item-list-content"
  >
    <div
      className="item-list-item"
      key="MCO854837339"
    >
      <Link
        to="/items/MCO854837339"
      >
        <Item
          item={
            Object {
              "condition": "new",
              "free_shipping": true,
              "id": "MCO854837339",
              "location": "Bogotá D.C.",
              "picture": "http://http2.mlstatic.com/D_613152-MLA47073710736_082021-I.jpg",
              "price": Object {
                "amount": 5555,
                "currency": "COP",
                "decimals": 0,
              },
              "title": "Impresora de buena calidad",
            }
          }
        />
        <div
          className="item-divider"
        />
      </Link>
    </div>
    <div
      className="item-list-item"
      key="MCO9999"
    >
      <Link
        to="/items/MCO9999"
      >
        <Item
          item={
            Object {
              "condition": "used",
              "free_shipping": false,
              "id": "MCO9999",
              "location": "Popayan",
              "picture": "http://http2.mlstatic.com/D_613152-MLA47073710736_082021-I.jpg",
              "price": Object {
                "amount": 2222,
                "currency": "COP",
                "decimals": 0,
              },
              "title": "Computador de segunda",
            }
          }
        />
        <div
          className="item-divider"
        />
      </Link>
    </div>
    <div
      className="item-list-item"
      key="MCO8547777"
    >
      <Link
        to="/items/MCO8547777"
      >
        <Item
          item={
            Object {
              "condition": "new",
              "free_shipping": true,
              "id": "MCO8547777",
              "location": "Bogotá D.C.",
              "picture": "http://http2.mlstatic.com/D_613152-MLA47073710736_082021-I.jpg",
              "price": Object {
                "amount": 111111,
                "currency": "COP",
                "decimals": 0,
              },
              "title": "Mouse de buena calidad",
            }
          }
        />
        <div
          className="item-divider"
        />
      </Link>
    </div>
    <div
      className="item-list-item"
      key="MCO999944444"
    >
      <Link
        to="/items/MCO999944444"
      >
        <Item
          item={
            Object {
              "condition": "used",
              "free_shipping": false,
              "id": "MCO999944444",
              "location": "Popayan",
              "picture": "http://http2.mlstatic.com/D_613152-MLA47073710736_082021-I.jpg",
              "price": Object {
                "amount": 9898989,
                "currency": "COP",
                "decimals": 0,
              },
              "title": "Diademas de segunda",
            }
          }
        />
        <div
          className="item-divider"
        />
      </Link>
    </div>
  </div>
</div>
`;
